source/unified-test-format/unified-test-format.rst

```diff
@@ -0,0 +1,1896 @@
+    { $$type: [ <bsonType>, <bsonType>, ... ] }
+
+This operator can be used anywhere a matched value is expected (including an
+`expectedResult <operation_expectedResult_>`_). The test runner MUST assert that
+the actual value exists and matches one of the expected types, which correspond
+to the documented types for the
+`$type <https://docs.mongodb.com/manual/reference/operator/query/type/>`__
+query operator.
+
+An example of this operator follows::
+
+    command:
+      getMore: { $$type: [ int, long ] }
+      collection: { $$type: 2 } # string
+
+When the actual value is an array, test runners MUST NOT examine types of the
+array's elements. Only the type of actual field should be checked. This is
+admittedly inconsistent with the behavior of the
+`$type <https://docs.mongodb.com/manual/reference/operator/query/type/>`__
+query operator, but there is presently no need for this behavior in tests.
+
+
+$$unsetOrMatches
+````````````````
+
+Syntax::
+
+    { $$unsetOrMatches: <anything> }
+
+This operator can be used anywhere a matched value is expected (including an
+`expectedResult <operation_expectedResult_>`_). The test runner MUST assert that
+actual value either does not exist or and matches the expected value. Matching
+the expected value should use the standard rules in `Evaluating Matches`_, which
+means that it may contain special operators.
+
+This operator is primarily used to assert driver-optional fields from the CRUD
+spec (e.g. ``insertedId`` for InsertOneResult, ``writeResult`` for
+BulkWriteException).
+
+An example of this operator used for a result's field follows::
+
+    expectedResult:
+      insertedId: { $$unsetOrMatches: 2 }
+
+An example of this operator used for an entire result follows::
+
+    expectedError:
+      expectedResult:
+        $$unsetOrMatches:
+          deletedCount: 0
+          insertedCount: 2
+          matchedCount: 0
+          modifiedCount: 0
+          upsertedCount: 0
+          upsertedIds: { }
+
+
+$$sessionLsid
+`````````````
+
+Syntax::
+
+    { $$sessionLsid: <string> }
+
+This operation is used for matching any value with the logical session ID of a
+`session entity <entity_session_>`_. The value will refer to a unique name of a
+session entity. The YAML file SHOULD use an `alias node`_ for a session entity's
+``id`` field (e.g. ``session: *session0``).
+
+An example of this operator follows::
+
+    command:
+      ping: 1
+      lsid: { $$sessionLsid: *session0 }
+
+
+Test Runner Implementation
+--------------------------
+
+The sections below describe instructions for instantiating the test runner,
+loading each test file, and executing each test within a test file. Test runners
+SHOULD NOT share state created by processing a test file with the processing of
+subsequent test files, and likewise for tests within a test file.
+
+
+Configuring the Test Runner
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The instructions in this section apply once for the test runner.
```

Comment:
What do you mean by 'apply once'?

Reply:
This was derived from the "apply once" lines to subsequent sections, but the wording can certainly be improved. The point I was trying to make is that these instructions should only be followed once when the test runner is instantiated.

Reply:
I think changing the title of this section to "Initializing the Test Runner" makes more sense, so I'll do that and delete this sentence.